src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -305,19 +308,40 @@ static void assertNoOverlap(InitCodeNode root, List<String> paths) {
     // that reference the same nodes.
     HashMap<InitCodeNode, String> refFrom = new HashMap<>();
 
+    // Keep track of the resource name entities. Configuring an entity twice or configuring an
+    // entity
+    // and the parent node at the same time will cause collision. Configuring two different entities
+    // will not.
+    Multimap<InitCodeNode, String> nodeEntities = HashMultimap.create();
+
     // Below we'll perform depth-first search, keep a list of nodes we've seen but have not
     // descended into. It doesn't really matter if we search breath- or depth-first; DFS is a little
     // more efficient on average.
     ArrayDeque<InitCodeNode> subNodes = new ArrayDeque<>();
 
     for (String path : paths) {
       subNodes.add(root.subTree(path));
+      String entity = FieldStructureParser.parseEntityName(path);
       while (!subNodes.isEmpty()) {
         InitCodeNode node = subNodes.pollLast();
         String oldPath = refFrom.put(node, path);
-        if (oldPath != null) {
-          throw new IllegalArgumentException(
-              String.format("SampleInitAttribute %s overlaps with %s", oldPath, path));
+        if (oldPath == null) {
+          checkArgument(
+              entity == null || nodeEntities.put(node, entity),
+              "SampleInitAttribute %s overlaps with %s",
+              path,
+              path);
+        } else {
+          checkArgument(
+              entity != null && nodeEntities.containsKey(node),
+              "SampleInitAttribute %s overlaps with %s",
+              oldPath,
+              path);
+          checkArgument(
+              nodeEntities.put(node, entity),
+              "SampleInitAttribute %s overlaps with %s",
```

Comment:
how about something ~`"entity '%s' in parameter '%s' specified multiple times", entity, node`